scripts/generate_bom.dart

```diff
@@ -0,0 +1,201 @@
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:intl/intl.dart';
+import 'package:pubspec_parse/pubspec_parse.dart' as pubspec;
+
+const packagesDir = 'packages';
+const versionsFile = 'VERSIONS.md';
+const versionsJsonFile = 'scripts/versions.json';
+const androidVersionFile =
+    '$packagesDir/firebase_core/firebase_core/android/gradle.properties';
+const iosVersionFile =
+    '$packagesDir/firebase_core/firebase_core/ios/firebase_sdk_version.rb';
+const webVersionFile =
+    '$packagesDir/firebase_core/firebase_core_web/lib/src/firebase_sdk_version.dart';
+const windowsVersionFile =
+    '$packagesDir/firebase_core/firebase_core/windows/CMakeLists.txt';
+
+const List<String> packages = [
+  'firebase_core',
+  'firebase_auth',
+  'cloud_firestore',
+  'firebase_storage',
+  'firebase_database',
+  'firebase_messaging',
+  'firebase_crashlytics',
+  'firebase_performance',
+  'firebase_remote_config',
+  'firebase_analytics',
+  'firebase_in_app_messaging',
+  'firebase_app_check',
+  'firebase_ml_model_downloader',
+];
+
+const jsonEncoder = JsonEncoder.withIndent('    ');
+
+void main(List<String> arguments) async {
+  stdout.write('BoM Version number: ');
+  String? version = stdin.readLineSync();
+  String date = DateFormat('yyyy-MM-dd').format(DateTime.now());
+
+  // Fetch native versions
+  String androidSdkVersion = await getSdkVersion(
+    r"awk -F '=' '/FirebaseSDKVersion/{print $2}' $VERSION_FILE",
+    androidVersionFile,
+  );
+  String iosSdkVersion = await getSdkVersion(
+    "awk -F \"'\" '/def firebase_sdk_version!()/ {getline; print \$2}' \$VERSION_FILE",
+    iosVersionFile,
+  );
+  String webSdkVersion = await getSdkVersion(
+    "awk -F \"'\" '/const String supportedFirebaseJsSdkVersion =/ {print \$2}' \$VERSION_FILE",
+    webVersionFile,
+  );
+  String windowsSdkVersion = await getSdkVersion(
+    "awk -F '\"' '/set\\(FIREBASE_SDK_VERSION/ {print \$2}' \$VERSION_FILE",
+    windowsVersionFile,
+  );
```

Comment:
we can just use file read and simple regexes to do this, 

Something like this should work (untested):

```dart
String androidPattern = r'FirebaseSDKVersion\s*=\s*\'([^\']+)\'';
String iosPattern = r'def firebase_sdk_version!()\s*\'([^\']+)\'';
String webPattern = r'const String supportedFirebaseJsSdkVersion = \'([^\']+)\'';
String windowsPattern = r'FIREBASE_SDK_VERSION\s*\"([^\"]+)\"';

Future<String> getSdkVersion(String pattern, String versionFile) async {
  RegExp regex = RegExp(pattern);
  String fileContents = await File(versionFile).readAsString();
  Match? match = regex.firstMatch(fileContents);
  return match?.group(1)?.trim() ?? "Version not found";
}

// ...
// Fetch native versions
String androidSdkVersion = await getSdkVersion(
  androidPattern,
  androidVersionFile,
);
String iosSdkVersion = await getSdkVersion(
  iosPattern,
  iosVersionFile,
);
String webSdkVersion = await getSdkVersion(
  webPattern,
  webVersionFile,
);
String windowsSdkVersion = await getSdkVersion(
  windowsPattern,
  windowsVersionFile,
);
```